docs/topics/native/native-improving-compilation-time.md

```diff
@@ -8,80 +10,150 @@ Read on for our tips on how to speed up the Kotlin/Native compilation process.
\ No newline at end of file
+Below are some recommendations for configuring Gradle for better compilation performance.
+
+### Increase Gradle heap size
+
+To increase the [Gradle heap size](https://docs.gradle.org/current/userguide/performance.html#adjust_the_daemons_heap_size),
+add `org.gradle.jvmargs=-Xmx3g` to your `gradle.properties` file.
+
+If you use [parallel builds](https://docs.gradle.org/current/userguide/performance.html#parallel_execution),
+you might need to choose the right number of workers with the `org.gradle.workers.max` property or the `--max-workers` command-line option.
+The default value is the number of CPU processors. 
+
+### Build only necessary binaries
+
+Don't run Gradle tasks that build the whole project, such as `build` or `assemble`, unless you really need to.
+These tasks build the same code more than once, increasing the compilation times. In typical cases, such as running tests
+from IntelliJ IDEA or starting the app from Xcode, the Kotlin tooling avoids executing unnecessary tasks. 
+
+If you have a non-typical case or build configuration, you might need to choose the task yourself:
+
+* `linkDebug*`. To run your code during development, you usually need only one binary, so running the corresponding
+  `linkDebug*` task should be enough.
+* `embedAndSignAppleFrameworkForXcode`. Since iOS simulators and devices have different processor architectures,
+  it's a common approach to distribute a Kotlin/Native binary as a universal (fat) framework.
+
+  However, during local development, it's faster to build the `.framework` file only for the platform you're using.
+  To build a platform-specific framework, use the [embedAndSignAppleFrameworkForXcode](multiplatform-direct-integration.md#connect-the-framework-to-your-project) task.
+
+### Build only for necessary targets
+
+Similarly to the recommendation above, don't build a binary for all native
+platforms at once. For example, compiling an [XCFramework](multiplatform-build-native-binaries.md#build-xcframeworks)
+(using an `*XCFramework` task) builds the same code for all targets, which takes proportionally more time than
+building for a single target.
+
+If you do need XCFrameworks for your setup, you can reduce the number of targets.
+For example, you don't need `iosX64` if you don't run this project on iOS simulators on Intel-based Macs.
+
+> Binaries for different targets are built with `linkDebug*$Target` and `linkRelease*$Target` Gradle tasks.
+> You can look for the executed tasks in the build log or in the
+> [Gradle build scan](https://docs.gradle.org/current/userguide/build_scans.html)
+> by running a Gradle build with the `--scan` option.
+>
+{style="tip"}
+
+### Don't build release binaries
```

Comment:
I think the "unless necessary" part of that heading was useful. Without it, this might sound like a recommendation to avoid release binaries completely, which is confusing. Instead, the message should be to avoid them during development, when performance doesn't matter that much. Apps should be released with release binaries in any case.